src/modules/tokens/hooks/useOnchainBalances.ts

```diff
@@ -98,7 +98,7 @@ function useOnchainErc20Amounts(
 
   // Return amounts
   return useMemo(() => {
-    if (!account || validatedTokens.length === 0) {
+    if (!account || validatedTokens.length === 0 || balancesCallState.length === 0) {
```

Comment:
This is new.
Why is that?

Reply:
Yes. Actually, it's a cause of https://github.com/cowprotocol/cowswap/issues/2705 issue.
With the new approach it should be a problem, but I left it there just in case